UserService/app/src/main/java/com/bookmyshow/userservice/service/RoleService.java

```diff
@@ -0,0 +1,135 @@
+package com.bookmyshow.userservice.service;
+
+import com.bookmyshow.userservice.dao.entity.Permission;
+import com.bookmyshow.userservice.dao.entity.Role;
+import com.bookmyshow.userservice.dao.repo.PermissionRepository;
+import com.bookmyshow.userservice.dao.repo.RoleRepository;
+import com.bookmyshow.userservice.dto.PermissionResponseDTO;
+import com.bookmyshow.userservice.dto.RoleRequestDTO;
+import com.bookmyshow.userservice.dto.RoleResponseDTO;
+import com.bookmyshow.userservice.exception.PermissionNotFoundException;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.bookmyshow.userservice.exception.RoleNotFoundException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class RoleService {
+
+    @Autowired
+    private ModelMapper modelMapper;
+
+    @Autowired
+    private RoleRepository repository;
+
+    @Autowired
+    private PermissionRepository permissionRepository;
+
+    public RoleResponseDTO addRole(RoleRequestDTO roleRequestDTO){
```

Comment:
Method names to be refactored as per the same comments provided for PermissionService.java